### ISSUE TITLE
Split pdf double pages and skew correction

### ISSUE THREAD

**Issue by noembryo** [koreader/koreader#6700]
It would be nice if there was a way to split double pages (when scanning both pages of an open book) in half.
In android there is EBookDroid app that it also uses MUPDF (I think) for pdfs, that has this and its really helpful.
We can also add a skew correction into the mix, because most of the time it is also needed..
I don't know if there is support for all this in MUPDF, but if it is.. ;o)

**Comment by Frenzie** [comment]
I don't think that's a feature in MuPDF, but k2pdfopt definitely has more options than is exposed. (But just plain reflow might do the trick?)

**Comment by noembryo** [comment]
>(But just plain reflow might do the trick?)
With some pdf (the most clean scanned) it does help, but with many the result is strange..
Some pages look OK some are really tangled..
It gets worst if there are columns and images in the pages..

**Comment by Galunid** [comment]
There is [autostraightening](https://www.willus.com/k2pdfopt/help/straighten.shtml) in k2optpdf, but I didn't see anything related in KOReader

**Comment by noembryo** [comment]
@Galunid That's exactly what I was thinking..
The  -m option (ignoring borders) is also a good feature! [hint-hint]

**Comment by Frenzie** [comment]
@Galunid That means you'd have to add it. :-P

**Comment by Galunid** [comment]
> The -m option (ignoring borders) is also a good feature! [hint-hint]

I believe you can achieve it already with page cropping ;-)



> @Galunid That means you'd have to add it. :-P

I'll take a look once I'm done with my other PRs

**Comment by noembryo** [comment]
>I believe you can achieve it already with page cropping ;-)
Not with auto cropping, you can't.
Specially when there is a dark part when a scanner scans a little outside of the page.
In that case auto crop crops nothing of the page..

You can do it with semi-auto cropping, but then if you reading a scanned book that the pages are not aligned (most of the manually scanned books are like that), you have to re-adjust the borders for every page :o(

**Comment by Galunid** [comment]
From what I understood, the "-m" option works by specifying top, bottom, left, right margins (they are constant), so you'd have to adjust those anyway. I don't think it works when the margins change. I assume first you'd have to deskew it, then split pages, then apply "-m" option, which in theory should be equivalent to autocropping.

TL;DR
k2pdfopt doesn't automatically detect page margins, so "-m" option in theory won't work when autocropping doesn't work (changing margins)

**Comment by noembryo** [comment]
Ah, OK then.
